src/charm.py

```diff
@@ -325,6 +325,8 @@ def config_changed(self) -> bool:
             )
             self.zookeeper_config.set_jaas_config()
 
+        self.zookeeper_config.set_server_jvmflags()
```

Comment:
nit: Looking at this, we would probably benefit from having some check on `src/utils.py -> update_env()` method. To avoid rewriting the file when there are no changes.